.github/workflows/linux_simd.yml

```diff
@@ -67,220 +65,147 @@ jobs:
+        sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-8 1
 
-  with_baseline_only:
-    needs: [smoke_test]
-    runs-on: ubuntu-latest
-    if: github.event_name != 'push'
-    env:
-      MESON_ARGS: "-Dallow-noblas=true -Dcpu-dispatch=none"
-    steps:
-    - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
-      with:
-        submodules: recursive
-        fetch-depth: 0
-    - uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4.7.0
-      with:
-        python-version: '3.10'
     - uses: ./.github/meson_actions
+      name: Build/Test against gcc-8
+
+    - name: Enable gcc-9
+      run: |
+        sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 1
+        sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-9 1
 
-  without_avx512:
-    needs: [smoke_test]
-    runs-on: ubuntu-latest
-    if: github.event_name != 'push'
-    env:
-      MESON_ARGS: "-Dallow-noblas=true -Dcpu-dispatch=SSSE3,SSE41,POPCNT,SSE42,AVX,F16C,AVX2,FMA3"
-    steps:
-    - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
-      with:
-        submodules: recursive
-        fetch-depth: 0
-    - uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4.7.0
-      with:
-        python-version: '3.10'
     - uses: ./.github/meson_actions
+      name: Build/Test against gcc-9
 
-  without_avx512_avx2_fma3:
-    needs: [smoke_test]
+  specialize:
+    needs: [baseline_only]
     runs-on: ubuntu-latest
     if: github.event_name != 'push'
+    continue-on-error: true
+    strategy:
+      matrix:
+        BUILD_PROP:
+          - [
+            "without optimizations",
+            "-Dallow-noblas=true -Ddisable-optimization=true",
+            "3.12-dev"
+          ]
+          - [
+            "native",
+            "-Dallow-noblas=true -Dcpu-baseline=native -Dcpu-dispatch=none",
+            "3.11"
+          ]
+          - [
+            "without avx512",
+            "-Dallow-noblas=true -Dcpu-dispatch=SSSE3,SSE41,POPCNT,SSE42,AVX,F16C,AVX2,FMA3",
+            "3.10"
+          ]
+          - [
+            "without avx512/avx2/fma3",
+            "-Dallow-noblas=true -Dcpu-dispatch=SSSE3,SSE41,POPCNT,SSE42,AVX,F16C",
+            "3.9"
+          ]
+
     env:
-      MESON_ARGS: "-Dallow-noblas=true -Dcpu-dispatch=SSSE3,SSE41,POPCNT,SSE42,AVX,F16C"
+      MESON_ARGS: ${{ matrix.BUILD_PROP[1] }}
+
+    name: "${{ matrix.BUILD_PROP[0] }}"
     steps:
     - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
       with:
         submodules: recursive
         fetch-depth: 0
     - uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4.7.0
       with:
-        python-version: '3.10'
+        python-version: "${{ matrix.BUILD_PROP[2] }}"
     - uses: ./.github/meson_actions
+      name: Build/Test
 
-  armv7_simd_test:
```

Comment:
For completeness: a superset of this test is now tested in `linux_qemu.yml`, so this removal is expected.